doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,115 +1257,82 @@ Managing the local context
    for any unresolved variables rather than failing.  Typically this happens when
    using a ``%`` intropattern (see :n:`@simple_intropattern`).
 
-.. tacn:: clear @ident
-   :name: clear
+.. tacn:: clear {| {? {? - } {+ @ident } } | @natural }
 
-   This tactic erases the hypothesis named :n:`@ident` in the local context of
-   the current goal. As a consequence, :n:`@ident` is no more displayed and no
-   more usable in the proof development.
+   Erases hypotheses from the context of the current goal, provided that they are
+   not referenced directly or indirectly from the remaining hypotheses or the goal.
```

Comment:
```suggestion
   not dependencies of the remaining hypotheses or the goal.
```

or something similar.

Reply:
I will tweak, but note that "dependencies of the remaining hypotheses" gets the relationship backward.  A dependency of H is something that H uses.

Reply:
If I have `a : A` in the context and my goal is `P a`, then isn't `a` a dependency of the goal?

Reply:
I made it:

   Erases hypotheses from the context of the current goal, provided that the
   remaining hypotheses and the goal don't depend directly or indirectly on the
   erased hypotheses.

Reply:
Looks good, but I think you can drop the "directly or indirectly".

Reply:
It's for the benefit of beginners.